codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/EndpointBasedAuthSchemeProviderSpec.java

```diff
@@ -123,7 +124,8 @@ private MethodSpec resolveAuthSchemeMethod() {
             }
         });
         spec.addStatement(".build()");
-        spec.addStatement("$T endpoint = DELEGATE.resolveEndpoint(endpointParameters).join()", Endpoint.class);
+        spec.addStatement("$T endpoint = $T.joinLikeSync(DELEGATE.resolveEndpoint(endpointParameters))",
+                          Endpoint.class, CompletableFutureUtils.class);
```

Comment:
Needed to fix some test cases in S3. It asserts on the exception type that's thrown, and that changed when we switched to the endpoint-based auth scheme providers without `joinLikeSync`.